crates/node/src/main.rs

```diff
@@ -133,15 +135,21 @@ fn generate_node_key(opts: NodeKeyOptions) -> Result<()> {
 
 #[async_trait]
 impl mempool::Storage for storage::Database {
-    async fn get(&self, hash: &Hash) -> Result<Option<Transaction>> {
+    async fn get(&self, hash: &Hash) -> Result<Option<Transaction<Validated>>> {
         self.find_transaction(hash).await
     }
 
-    async fn set(&self, tx: &Transaction) -> Result<()> {
-        self.add_transaction(tx).await
+    async fn set(&self, tx: &Transaction<Validated>) -> Result<()> {
+        let tx_hash = tx.hash;
+        self.add_transaction(tx).await?;
+        self.add_asset(&tx_hash).await?;
+        self.mark_asset_complete(&tx_hash).await
```

Comment:
I remove the asset manager because its role was taken by the validation process. For the database, I don't know. It depends on why asset are store in the db. If we need to keep trace of them (history) we should keep it. If we only need the file part, describe in the Tx data, they are not needed.
I think we should keep them for now until we define the Tx execution and storage domains.